AzureIotPnpDps.X/main.c

```diff
@@ -7,582 +7,1327 @@
-  }
+/**********************************************
+* Add JSON for writable property response with int32 data
+* e.g. "property_name" : property_val_int32
+**********************************************/
+static az_result append_reported_property_response_int32(
+	az_json_writer* jw,
+	az_span property_name,
+	int32_t property_val,
+	int32_t ack_code,
+	int32_t ack_version,
+	az_span ack_description)
+{
+	RETURN_ERR_IF_FAILED(az_json_writer_append_property_name(jw, property_name));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_begin_object(jw));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_property_name(jw, span_desired_ack_value));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_int32(jw, property_val));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_property_name(jw, span_desired_ack_code));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_int32(jw, ack_code));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_property_name(jw, span_desired_ack_version));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_int32(jw, ack_version));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_property_name(jw, span_desired_ack_description));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_string(jw, ack_description));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_end_object(jw));
+	return AZ_OK;
+}
 
-  return rc;
+/**********************************************
+*	Add JSON key-value pairs for getMaxMinReport command
+*	e.g.
+*	{
+*		"maxTemp" : 0,
+*		"minTemp" : 2,
+*		"minTemp" : 1,
+*		"startTime" : 1,
+*		"endTime" : 1,
+*	}
+**********************************************/
+static az_result build_getMaxMinReport_response_payload(
+	az_span response_span,
+	az_span start_time_span,
+	az_span end_time_span,
+	az_span* response_payload_span)
+{
+	az_json_writer jw;
+
+	// Build the command response payload
+	RETURN_ERR_IF_FAILED(start_json_object(&jw, response_span));
+
+	RETURN_ERR_IF_FAILED(append_jason_property_int32(&jw, span_command_getmaxmin_max_temp, device_max_temp));
+	RETURN_ERR_IF_FAILED(append_jason_property_int32(&jw, span_command_getmaxmin_min_temp, device_min_temp));
+	RETURN_ERR_IF_FAILED(append_jason_property_int32(&jw, span_command_getmaxmin_avg_temp, device_avg_temp));
+	RETURN_ERR_IF_FAILED(append_jason_property_string(&jw, span_command_getmaxmin_start_time, start_time_span));
+	RETURN_ERR_IF_FAILED(append_jason_property_string(&jw, span_command_getmaxmin_end_time, end_time_span));
+	RETURN_ERR_IF_FAILED(end_json_object(&jw));
+	*response_payload_span = az_json_writer_get_bytes_used_in_destination(&jw);
+	return AZ_OK;
 }
 
-static az_result build_command_response_payload(
-    az_json_writer* json_writer,
-    az_span start_time_span,
-    az_span end_time_span,
-    az_span* response_payload)
+/**********************************************
+* Create JSON document for error response
+**********************************************/
+static az_result build_error_response_payload(
+	az_span response,
+	az_span error_string,
+	az_span* response_payload)
```

Comment:
above function has `_span` as a suffix. consider:
```suggestion
	az_span response_span,
	az_span error_string_span,
	az_span* response_payload_span)
```

Reply:
Replaced all variables with _span suffix

[commit](https://github.com/daisukeiot/Microchip-PIC-IoT-Wx-1/commit/64508ae802742fd040c4263817d02b3a719ddeaa)